healthcheck/healthcheck.go

```diff
@@ -150,20 +150,52 @@ type Observation struct {
 	// Backoff is the amount of time we back off between retries for failed
 	// checks.
 	Backoff time.Duration
+
+	// OnSuccess is a callback which will be executed when the healthcheck
+	// succeeds. This is optional.
+	OnSuccess func()
+
+	// OnFailure is a callback which will be executed when the healthcheck
+	// fails. This is optional.
+	OnFailure func()
+}
+
+type ObservationOption func(*Observation)
+
+func WithSuccessCallback(callback func()) ObservationOption {
+	return func(o *Observation) {
+		o.OnSuccess = callback
+	}
+}
+
+func WithFailureCallback(callback func()) ObservationOption {
+	return func(o *Observation) {
+		o.OnFailure = callback
+	}
 }
 
 // NewObservation creates an observation.
-func NewObservation(name string, check func() error, interval,
-	timeout, backoff time.Duration, attempts int) *Observation {
-
-	return &Observation{
-		Name:     name,
-		Check:    CreateCheck(check),
-		Interval: ticker.New(interval),
-		Attempts: attempts,
-		Timeout:  timeout,
-		Backoff:  backoff,
+func NewObservation(name string, check func() error,
+	interval, timeout, backoff time.Duration, attempts int,
+	opts ...ObservationOption) *Observation {
+
+	observation := &Observation{
+		Name:      name,
+		Check:     CreateCheck(check),
+		Interval:  ticker.New(interval),
+		Attempts:  attempts,
+		Timeout:   timeout,
+		Backoff:   backoff,
+		OnSuccess: func() {}, // Default NO-OP success callback
```

Comment:
nit: could create a `NoOpCallback` function to avoid the need for this inline comment.

Reply:
✅